db/migration/1653646292637-ArchiveDatasets.ts

```diff
@@ -0,0 +1,190 @@
+    1641, 1642, 1643, 1644, 1645, 1646, 1647, 1648, 1649, 1650, 1651, 1652,
+    1653, 1654, 1655, 1656, 1657, 1658, 1659, 1660, 1661, 1662, 1663, 1664,
+    1665, 1666, 1667, 1668, 1669, 1670, 1671, 1672, 1673, 1674, 1675, 1676,
+    1677, 1678, 1679, 1680, 1681, 1682, 1683, 1684, 1685, 1686, 1687, 1688,
+    1689, 1690, 1691, 1692, 1693, 1694, 1695, 1696, 1697, 1698, 1699, 1700,
+    1701, 1702, 1703, 1704, 1705, 1706, 1707, 1708, 1709, 1710, 1711, 1712,
+    1713, 1714, 1715, 1716, 1717, 1718, 1719, 1720, 1721, 1722, 1723, 1724,
+    1725, 1726, 1727, 1728, 1729, 1730, 1731, 1732, 1733, 1734, 1735, 1736,
+    1737, 1738, 1739, 1742, 1743, 1744, 1745, 1746, 1747, 1748, 1749, 1750,
+    1751, 1752, 1753, 1754, 1755, 1756, 1757, 1758, 1759, 1760, 1761, 1762,
+    1763, 1764, 1765, 1766, 1767, 1768, 1769, 1770, 1771, 1772, 1773, 1774,
+    1775, 1776, 1777, 1778, 1779, 1780, 1781, 1782, 1783, 1784, 1785, 1786,
+    1787, 1788, 1789, 1790, 1791, 1792, 1793, 1794, 1795, 1796, 1797, 1798,
+    1799, 1800, 1801, 1802, 1803, 1804, 1805, 1806, 1807, 1808, 1809, 1810,
+    1811, 1812, 1813, 1814, 1815, 1816, 1817, 1818, 1819, 1820, 1821, 1822,
+    1823, 1824, 1825, 1826, 1827, 1828, 1829, 1830, 1045, 1048, 1058, 1060,
+    1063, 1076, 1077, 1078, 1079, 1082, 1084, 1088, 1055, 1059, 1066, 1074,
+    1075, 1090, 1081, 1044, 1054, 1064, 1073, 1042, 1043, 1046, 1050, 1057,
+    1061, 1062, 1065, 1083, 1085, 1089, 1086, 1051, 1052, 1067, 1030, 973, 969,
+    932, 927, 922, 923, 878, 880, 881, 886, 898, 899, 900, 905, 879, 887, 888,
+    889, 890, 891, 892, 893, 894, 895, 896, 906, 909, 910, 912, 914, 915, 882,
+    883, 884, 885, 903, 904, 907, 908, 911, 901, 902, 913, 897, 843, 776, 779,
+    782, 784, 786, 787, 791, 793, 794, 795, 796, 799, 800, 801, 804, 806, 807,
+    810, 813, 814, 815, 816, 817, 820, 821, 824, 825, 826, 827, 828, 831, 832,
+    833, 835, 836, 837, 838, 839, 788, 798, 809, 818, 777, 790, 812, 823, 830,
+    803, 802, 808, 819, 781, 811, 778, 783, 789, 840, 841, 797, 822, 834, 696,
+    669, 645, 637, 618, 617, 616, 628, 630, 624, 627, 620, 626, 619, 621, 634,
+    635, 636, 638, 593, 556, 557, 558, 559, 417, 2997, 350, 306, 280, 207, 2980,
+    153, 137, 2979,
+]
+
+export class ArchiveDatasets1653646292637 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await archiveDatasets(queryRunner, DATASETS_TO_ARCHIVE)
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        await archiveDatasets(queryRunner, DATASETS_TO_ARCHIVE, true)
+    }
+}
+
+const archiveDatasets = async (
+    queryRunner: QueryRunner,
+    datasetIds: number[],
+    revert: boolean = false
+) => {
+    // label datasets as archived in DB
+    const placeholders = datasetIds.map(() => "?")
+
+    await queryRunner.query(
+        `UPDATE datasets SET isArchived = ${revert ? 0 : 1}
+        WHERE id in (${placeholders})`,
+        datasetIds
+    )
```

Comment:
nit: I think there is no need for placeholders, you can pass an array as a parameter, like here: https://github.com/owid/owid-grapher/blob/9a94b86554e6782c94e624352553b5e29008ebfb/baker/siteRenderers.tsx#L557-L560

I think this should work:

```suggestion
    const placeholders = datasetIds.map(() => "?")

    await queryRunner.query(
        `UPDATE datasets SET isArchived = ${revert ? 0 : 1}
        WHERE id in (?)`,
        [datasetIds]
    )
```